pkg/machine/qemu/machine.go

```diff
@@ -661,7 +661,7 @@ func (v *MachineVM) Start(name string, opts machine.StartOptions) error {
 			return err
 		}
 		listening := v.isListening()
-		for state != machine.Running || !listening {
```

Comment:
Tried this out. Every time I've been able to reproduce this error, listening after line 633 is true, which means that usually this condition isn't met and the loop is never used, even on failures.